core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsConfig.scala

```diff
@@ -0,0 +1,137 @@
+ * <ul>
+ *   <li> subscriptionId: Name/ID supplied by CLI during the creation of the client metric subscription.
+ *   <li> subscribedMetrics: List of metric prefixes
+ *   <li> pushIntervalMs: A positive integer value >=0  tells the client that how often a client can push the metrics
+ *   <li> matchingPatternsList: List of client matching patterns, that are used by broker to match the client instance
+ *   with the subscription.
+ * </ul>
+ * }
+ * <p>
+ * At present, CLI can pass the following parameters in request to add/delete/update the client metrics
+ * subscription:
+ * <ul>
+ *  <li> "metrics" value should be comma separated metrics list. An empty list means no metrics subscribed.
+ *      A list containing just an empty string means all metrics subscribed.
+ *      Ex: "org.apache.kafka.producer.partition.queue.,org.apache.kafka.producer.partition.latency"
+ *  <li> "interval.ms" should be between 100 and 3600000 (1 hour). This is the interval at which the client
+ *      should push the metrics to the broker.
+ *  <li> "match" is a comma separated list of client match patterns, in case if there is no matching
+ *      pattern specified then broker considers that as all match which means the associated metrics
+ *      applies to all the clients. Ex: "client_software_name = Java, client_software_version = 11.1.*"
+ * </ul>
+ * For more information please look at kip-714:
+ * https://cwiki.apache.org/confluence/display/KAFKA/KIP-714%3A+Client+metrics+and+observability#KIP714:Clientmetricsandobservability-Clientmetricsconfiguration
+ */
+object ClientMetricsConfig {
+
+  class SubscriptionInfo(subscriptionId: String,
+                         subscribedMetrics: List[String],
+                         pushIntervalMs: Int,
+                         var matchingPatternsList: List[String]) {
+    def getId: String = subscriptionId
+    def getPushIntervalMs: Int = pushIntervalMs
+    def getClientMatchingPatterns: Map[String, String] = ClientMetricsMetadata.parseMatchingPatterns(matchingPatternsList)
+    def getSubscribedMetrics: List[String] = subscribedMetrics
+  }
+
+  object ClientMatchingParams {
+    val CLIENT_ID = "client_id"
+    val CLIENT_INSTANCE_ID = "client_instance_id"
+    val CLIENT_SOFTWARE_NAME = "client_software_name"
+    val CLIENT_SOFTWARE_VERSION = "client_software_version"
+    val CLIENT_SOURCE_ADDRESS = "client_source_address"
+    val CLIENT_SOURCE_PORT = "client_source_port"
+
+    val matchersList = List(CLIENT_ID, CLIENT_INSTANCE_ID, CLIENT_SOFTWARE_NAME,
+                            CLIENT_SOFTWARE_VERSION, CLIENT_SOURCE_ADDRESS, CLIENT_SOURCE_PORT)
+
+    def isValidParam(param: String) = matchersList.contains(param)
+  }
+
+  object ClientMetrics {
+    // Properties that are used to create the subscription for client_metrics.
+    val SubscriptionMetrics = "metrics"
+    val PushIntervalMs = "interval.ms"
+    val ClientMatchPattern = "match"
+
+    val DEFAULT_PUSH_INTERVAL = 30 * 1000 // 5 minutes
```

Comment:
I would call it `DEFAULT_INTERVAL_MS` so it's clear it's milliseconds. Also, the config is `interval.ms`.

Reply:
Done.